rust/lancedb/src/index/vector.rs

```diff
@@ -201,3 +205,121 @@ pub(crate) fn suggested_num_sub_vectors(dim: u32) -> u32 {
         1
     }
 }
+
+/// Builder for an IVF_HNSW_SQ index.
+///
+/// This index is a combination of IVF and HNSW.
+/// The IVF part is the same as the IVF PQ index.
+/// For each IVF partition, this builds a HNSW graph, the graph is used to
+/// quickly find the closest vectors to a query vector.
+///
+/// The SQ (scalar quantizer) is used to compress the vectors,
+/// each vector is mapped to a 8-bit integer vector, 4x compression ratio for float32 vector.
+#[derive(Debug, Clone)]
+pub struct IvfHnswSqIndexBuilder {
+    // IVF
+    pub(crate) distance_type: DistanceType,
+    pub(crate) num_partitions: Option<u32>,
+    pub(crate) sample_rate: u32,
+    pub(crate) max_iterations: u32,
+
+    // HNSW
+    pub(crate) m: usize,
+    pub(crate) ef_construction: usize,
```

Comment:
Can we use `u64` instead of `usize`?  We can convert to `usize` internally.  I'd rather avoid `usize` in our public API since it is slightly more confusing to think about.

Reply:
sure, let me change it to `u32` cause we don't need such great value of `u64`